### ISSUE TITLE
New Rule: no-duplicate-defined-urls

### ISSUE THREAD

**Issue by sethamus** [eslint/markdown#701]
### Rule details

Warn when the same URL is defined with different identifiers.

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
[mercury]: https://example.com/mercury/
[venus]: https://example.com/mercury/
```

### Prior Art

[remark-lint-no-duplicate-defined-urls](https://github.com/remarkjs/remark-lint/tree/main/packages/remark-lint-no-duplicate-defined-urls)

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by DMartens** [comment]
I agree that the rule would be helpful but I think the existing `no-duplicate-definitions` should be extended with an option to check for duplicate URLs.
Would like other opinions from the team.

**Comment by lumirlumir** [comment]
I’m in favor of this rule.

> I think the existing `no-duplicate-definitions` rule should be extended with an option to check for duplicate URLs.

I’d prefer this rule to be separate from `no-duplicate-definitions`. Although the node listeners would be nearly identical, the internal logic would need to account for additional considerations, and the purposes of this proposed rule and `no-duplicate-definitions` are somewhat different.

I also think this rule should be extended to support footnote definition syntax, so its internal logic would need to account for the structure of footnote definition children, similar to the logic in:

https://github.com/eslint/markdown/blob/ac31775656f65f78bc02c0a377c0a85bd131f9f8/src/rules/no-duplicate-headings.js#L95

If this rule supports footnote definitions, I think its name should be changed, since the current name implies that it only applies to URLs.

My suggestion would be `no-duplicate-definition-content`, but I’m open to other suggestions.

**Comment by DMartens** [comment]
@lumirlumir can you please specify what additional considerations you mean?
I do not hink footnotes should be a part of this rule. This would make more sense in a separate rule, e.g. `no-duplicate-text` as the same logic would also be helpful for list items or paragraphs.

**Comment by lumirlumir** [comment]
> I do not hink footnotes should be a part of this rule.

There’s a history of the existing definition-only checking rule being extended to support footnote definitions (https://github.com/eslint/markdown/pull/442). The three existing definition-related rules—`no-duplicate-definitions`, `no-empty-definitions`, and `no-unnecessary-definitions`—all check both `definition` and `footnoteDefinitions` within a single rule using the `checkFootnoteDefinitions` option. Separating this rule would therefore be inconsistent with the existing rules.

Also, if we name the rule `no-duplicate-defined-urls`, it could be limited to URLs and might cause problems if a situation like https://github.com/eslint/eslint/issues/21113#issuecomment-5151801004 occurs again.

> can you please specify what additional considerations you mean?

This assumes that we extend the rule to support footnote definitions, since the existing `no-duplicate-definitions` rule also checks footnote definitions.

The existing logic in `no-duplicate-headings` requires three node listeners to gather the structure of its children. The rule would need to be extended with node listeners like those below to correctly gather the sequence of its children without re-traversing them by accessing `node.children`, since a `footnoteDefinition` can contain various types of child nodes, unlike a `definition` node.

https://github.com/eslint/markdown/blob/ac31775656f65f78bc02c0a377c0a85bd131f9f8/src/rules/no-duplicate-headings.js#L71

https://github.com/eslint/markdown/blob/ac31775656f65f78bc02c0a377c0a85bd131f9f8/src/rules/no-duplicate-headings.js#L93

https://github.com/eslint/markdown/blob/ac31775656f65f78bc02c0a377c0a85bd131f9f8/src/rules/no-duplicate-headings.js#L105

**Comment by DMartens** [comment]
Your explanation makes sense to me.
As we waited long enough for other team members to chime in, I think we should go with your proposal of creating a new rule `no-duplicate-definition-content` handling both definition types.
@sethamus Are you still up for providing a PR for this considering the change of implementation?
